### PR TITLE
Cleanup of namespaces in parquet code.

### PR DESCRIPTION
**Pull request by nvdbaranec**
Cleans up several issues in the parquet code:

- We were using the namespace `cudf::io::detail::parquet`, when `cudf::io::parquet::detail` makes more sense. 
- Converts the `cudf::io::parquet::gpu` namespace to also just use `cudf::io::parquet::detail`
- Several detail-style headers and source files were using `cudf::io::parquet` when they should probably have been in the detail namespace. 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]

**Review by PointKernel — APPROVED** [review]
LGTM

File: cpp/src/io/functions.cpp [hunk 1]
```diff
@@ -470,8 +470,8 @@ void orc_chunked_writer::close()
   writer->close();
 }
 
-using namespace cudf::io::detail::parquet;
-namespace detail_parquet = cudf::io::detail::parquet;
+using namespace cudf::io::parquet::detail;
+namespace detail_parquet = cudf::io::parquet::detail;
```
Comment (vuule) [inline cpp/src/io/functions.cpp:474]: I think this only exists to disambiguate between reader/writer data members and reader/writer classes. Kind of odd, IMO. Maybe a battle for another day.

File: cpp/include/cudf/io/parquet.hpp [hunk 2]
```diff
@@ -1750,7 +1750,7 @@ class parquet_chunked_writer {
     std::vector<std::string> const& column_chunks_file_paths = {});
 
   /// Unique pointer to impl writer class
-  std::unique_ptr<cudf::io::detail::parquet::writer> writer;
+  std::unique_ptr<cudf::io::parquet::detail::writer> writer;
```
Comment (vuule) [inline cpp/include/cudf/io/parquet.hpp:1753]: would this be weird?
```suggestion
  std::unique_ptr<parquet::detail::writer> writer;
```

File: cpp/src/io/parquet/predicate_pushdown.cpp [hunk 3]
```diff
@@ -62,13 +62,17 @@ struct stats_caster {
 
   // uses storage type as T
   template <typename T, CUDF_ENABLE_IF(cudf::is_dictionary<T>() or cudf::is_nested<T>())>
-  static T convert(uint8_t const* stats_val, size_t stats_size, cudf::io::parquet::Type const type)
+  static T convert(uint8_t const* stats_val,
+                   size_t stats_size,
+                   cudf::io::parquet::detail::Type const type)
```
Comment (vuule) [inline cpp/src/io/parquet/predicate_pushdown.cpp:67]: 5X
```suggestion
                   Type const type)
```

File: cpp/src/io/utilities/column_buffer.cpp [hunk 4]
```diff
@@ -51,19 +51,21 @@ std::unique_ptr<column> gather_column_buffer::make_string_column_impl(rmm::cuda_
   return make_strings_column(*_strings, stream, _mr);
 }
 
-void inline_column_buffer::allocate_strings_data(rmm::cuda_stream_view stream)
+void cudf::io::detail::inline_column_buffer::allocate_strings_data(rmm::cuda_stream_view stream)
```
Comment (vuule) [inline cpp/src/io/utilities/column_buffer.cpp:54]: not sure why this was added
```suggestion
void inline_column_buffer::allocate_strings_data(rmm::cuda_stream_view stream)
```

File: cpp/src/io/utilities/column_buffer.cpp [hunk 5]
```diff
@@ -51,19 +51,21 @@ std::unique_ptr<column> gather_column_buffer::make_string_column_impl(rmm::cuda_
...
 {
   CUDF_EXPECTS(type.id() == type_id::STRING, "allocate_strings_data called for non-string column");
   // size + 1 for final offset. _string_data will be initialized later.
   _data = create_data(data_type{type_id::INT32}, size + 1, stream, _mr);
 }
 
-void inline_column_buffer::create_string_data(size_t num_bytes, rmm::cuda_stream_view stream)
+void cudf::io::detail::inline_column_buffer::create_string_data(size_t num_bytes,
```
Comment (vuule) [inline cpp/src/io/utilities/column_buffer.cpp:61]:
```suggestion
void inline_column_buffer::create_string_data(size_t num_bytes,
```

File: cpp/src/io/utilities/column_buffer.cpp [hunk 6]
```diff
@@ -324,7 +326,7 @@ std::unique_ptr<column> empty_like(column_buffer_base<string_policy>& buffer,
 }
 
 using pointer_type = gather_column_buffer;
-using string_type  = inline_column_buffer;
+using string_type  = cudf::io::detail::inline_column_buffer;
```
Comment (vuule) [inline cpp/src/io/utilities/column_buffer.cpp:329]:
```suggestion
using string_type  = inline_column_buffer;
```

File: cpp/src/io/utilities/column_buffer.cpp [hunk 7]
```diff
@@ -51,19 +51,21 @@ std::unique_ptr<column> gather_column_buffer::make_string_column_impl(rmm::cuda_
...
 
-void inline_column_buffer::create_string_data(size_t num_bytes, rmm::cuda_stream_view stream)
+void cudf::io::detail::inline_column_buffer::create_string_data(size_t num_bytes,
+                                                                rmm::cuda_stream_view stream)
 {
   _string_data = rmm::device_buffer(num_bytes, stream, _mr);
 }
 
-std::unique_ptr<column> inline_column_buffer::make_string_column_impl(rmm::cuda_stream_view stream)
+std::unique_ptr<column> cudf::io::detail::inline_column_buffer::make_string_column_impl(
```
Comment (vuule) [inline cpp/src/io/utilities/column_buffer.cpp:67]:
```suggestion
std::unique_ptr<column> inline_column_buffer::make_string_column_impl(
```

**Review by vuule — CHANGES_REQUESTED** [review]
Thank you for taking up this task. Gets rid of so many `gpu::`!

File: cpp/src/io/utilities/column_buffer.cpp (hunk 4 above)
Comment (nvdbaranec) [reply cpp/src/io/utilities/column_buffer.cpp:54]: Yeah that was me being a little quick with the copy/paste.

File: cpp/src/io/parquet/parquet_common.hpp [hunk 8]
```diff
@@ -21,6 +21,8 @@
 namespace cudf {
 namespace io {
 namespace parquet {
+namespace detail {
```
[the remark this answers is not in the document]
Comment (nvdbaranec) [reply cpp/src/io/parquet/parquet_common.hpp]: Done

**Review by vuule — APPROVED** [review]

**Pull request merged**